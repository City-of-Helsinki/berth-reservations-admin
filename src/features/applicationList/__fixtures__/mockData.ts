import { BERTH_APPLICATIONS } from '../__generated__/BERTH_APPLICATIONS';
import { getBerthApplicationData } from '../utils';
import { ApplicationStatus, LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: BERTH_APPLICATIONS = {
  berthApplications: {
    __typename: 'BerthApplicationNodeConnection',
    count: 3,
    edges: [
      // Full structure
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '1',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-0',
            pier: 'A',
            reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano',
          boatDraught: 5,
          boatRegistrationNumber: 'A 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Matti',
          harborChoices: [{ __typename: 'HarborChoiceType', harbor: '1', harborName: '1', priority: 0 }],
          id: 'MOCK-APPLICATION-0',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-0',
                    properties: { __typename: 'HarborProperties', name: 'Testisatama' },
                  },
                  identifier: 'G',
                },
              },
            },
            id: 'MOCK-LEASE-0',
            status: LeaseStatus.OFFERED,
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Most structure and nulled fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: {
            __typename: 'BerthSwitchType',
            berthNumber: '2',
            harbor: 'MOCK-HARBOR-0',
            harborName: 'Testisatama',
            id: 'MOCK-BERTH-SWITCH-1',
            pier: 'A',
            reason: null,
          },
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano II',
          boatDraught: 5,
          boatRegistrationNumber: 'B 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE' },
          firstName: 'Maija',
          harborChoices: null,
          id: 'MOCK-APPLICATION-1',
          lastName: 'Meikäläinen',
          lease: {
            __typename: 'BerthLeaseNode',
            id: 'MOCK-LEASE-1',
            status: LeaseStatus.OFFERED,
            berth: {
              __typename: 'BerthNode',
              number: '7',
              pier: {
                __typename: 'PierNode',
                properties: {
                  __typename: 'PierProperties',
                  identifier: 'Pier Brosnan',
                  harbor: {
                    __typename: 'HarborNode',
                    id: 'MOCK-HARBOR-1',
                    properties: {
                      __typename: 'HarborProperties',
                      name: null,
                    },
                  },
                },
              },
            },
          },
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
      // Minimum fields
      {
        __typename: 'BerthApplicationNodeEdge',
        node: {
          __typename: 'BerthApplicationNode',
          accessibilityRequired: false,
          berthSwitch: null,
          boatLength: 6,
          boatModel: 'Marine',
          boatName: 'Cama la Yano 3-D',
          boatDraught: 5,
          boatRegistrationNumber: 'C 12345',
          boatType: '1',
          boatWeight: 5,
          boatWidth: 0,
          createdAt: '2020-07-16',
          customer: { __typename: 'ProfileNode', id: 'MOCK-PROFILE-0' },
          firstName: 'Matti',
          harborChoices: [
            { __typename: 'HarborChoiceType', harbor: 'MOCK-HARBOR-0', harborName: 'Testisatama', priority: 0 },
          ],
          id: 'MOCK-APPLICATION-2',
          lastName: 'Möttönen',
          lease: null,
          municipality: 'Helsinki',
          status: ApplicationStatus.PENDING,
        },
      },
    ],
  },
  boatTypes: [{ __typename: 'BoatTypeType', id: '1', name: 'Purjevene' }],
};

export const mockTableData = getBerthApplicationData(mockData);
