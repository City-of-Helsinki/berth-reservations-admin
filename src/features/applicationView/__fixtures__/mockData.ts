import {
  INDIVIDUAL_APPLICATION,
  INDIVIDUAL_APPLICATION_additionalProducts as ADDITIONAL_PRODUCTS,
  INDIVIDUAL_APPLICATION_berthApplication as BERTH_APPLICATION,
  INDIVIDUAL_APPLICATION_berthApplication_berthSwitch as BERTH_SWITCH,
  INDIVIDUAL_APPLICATION_berthApplication_customer as CUSTOMER,
  INDIVIDUAL_APPLICATION_berthApplication_lease as LEASE,
  INDIVIDUAL_APPLICATION_boatTypes as BOAT_TYPES,
} from '../__generated__/INDIVIDUAL_APPLICATION';
import {
  AdditionalProductType,
  ApplicationStatus,
  BerthApplicationLanguage,
  BerthMooringType,
  CustomerGroup,
  InvoicingType,
  Language,
  PeriodType,
  PriceUnits,
  ProductServiceType,
} from '../../../@types/__generated__/globalTypes';

export const mockCustomer: CUSTOMER = {
  __typename: 'ProfileNode',
  comment: 'Comment',
  firstName: 'Testi',
  id: 'MOCK-CUSTOMER',
  invoicingType: InvoicingType.DIGITAL_INVOICE,
  language: Language.FINNISH,
  lastName: 'Testinen',
  organization: null,
  primaryAddress: { __typename: 'AddressNode', address: 'Testikatu 1', city: 'Helsinki', postalCode: '00100' },
  primaryEmail: { __typename: 'EmailNode', email: 'test@example.com' },
  primaryPhone: { __typename: 'PhoneNode', phone: '+358 00 000 0000' },
  customerGroup: CustomerGroup.PRIVATE,
};

export const mockBerthSwitch: BERTH_SWITCH = {
  __typename: 'BerthSwitchType',
  berthNumber: '1',
  harbor: 'MOCK-HARBOR',
  harborName: 'Test Harbor',
  id: 'MOCK-BERTH-SWITCH',
  pier: 'A',
  reason: { __typename: 'BerthSwitchReasonType', title: 'Reason' },
};

export const mockLease: LEASE = {
  __typename: 'BerthLeaseNode',
  berth: {
    __typename: 'BerthNode',
    comment: '',
    depth: null,
    isAccessible: null,
    length: 5,
    mooringType: BerthMooringType.DINGHY_PLACE,
    number: 1,
    pier: {
      __typename: 'PierNode',
      properties: {
        __typename: 'PierProperties',
        electricity: false,
        gate: false,
        harbor: {
          __typename: 'HarborNode',
          id: 'MOCK-HARBOR',
          properties: { __typename: 'HarborProperties', name: 'Test Harbor' },
        },
        identifier: 'A',
        lighting: false,
        mooring: false,
        wasteCollection: false,
        water: false,
      },
    },
    width: 0,
  },
  id: 'MOCK-LEASE',
  order: null,
};

const additionalProducts: ADDITIONAL_PRODUCTS = {
  __typename: 'AdditionalProductNodeConnection',
  edges: [
    {
      __typename: 'AdditionalProductNodeEdge',
      node: {
        __typename: 'AdditionalProductNode',
        id: 'MOCK-ADDITIONAL-PRODUCT-NODE',
        period: PeriodType.MONTH,
        priceUnit: PriceUnits.AMOUNT,
        priceValue: '25.00',
        productType: AdditionalProductType.OPTIONAL_SERVICE,
        service: ProductServiceType.DINGHY_PLACE,
      },
    },
  ],
};

const berthApplication: BERTH_APPLICATION = {
  __typename: 'BerthApplicationNode',
  accessibilityRequired: false,
  address: 'Testikatu 1',
  berthSwitch: null,
  boatDraught: 1,
  boatLength: 1,
  boatModel: 'Marine',
  boatName: 'Cama la Yano',
  boatRegistrationNumber: '1234',
  boatType: '7',
  boatWeight: 120,
  boatWidth: 1,
  businessId: '',
  companyName: '',
  createdAt: '2020-07-10T08:04:36.062819+00:00',
  customer: null,
  email: 'test@example.com',
  firstName: 'Testi',
  harborChoices: [
    { harbor: 'MOCK-HARBOR-B', priority: 1, harborName: 'Saukonpaaden venesatama', __typename: 'HarborChoiceType' },
  ],
  id: 'MOCK-APPLICATION',
  language: BerthApplicationLanguage.FI,
  lastName: 'Testinen',
  lease: null,
  municipality: 'Helsinki',
  phoneNumber: '+358 00 000 0000',
  status: ApplicationStatus.PENDING,
  zipCode: '00100',
  acceptBoatingNewsletter: true,
  acceptFitnessNews: false,
  acceptLibraryNews: false,
  acceptOtherCultureNews: true,
  applicationCode: '1234',
};

const boatTypes: BOAT_TYPES[] = [
  { id: '1', name: 'Jollavene', __typename: 'BoatTypeType' },
  { id: '2', name: 'Soutuvene', __typename: 'BoatTypeType' },
  { id: '3', name: 'Perämoottorivene', __typename: 'BoatTypeType' },
  { id: '4', name: 'Sisäperämoottorivene', __typename: 'BoatTypeType' },
  { id: '5', name: 'Keskimoottorivene', __typename: 'BoatTypeType' },
  { id: '6', name: 'Purjevene / moottoripursi', __typename: 'BoatTypeType' },
  { id: '7', name: 'Troolari', __typename: 'BoatTypeType' },
  { id: '8', name: 'Suuri alus (yli 20t)', __typename: 'BoatTypeType' },
];

export const mockApplication: INDIVIDUAL_APPLICATION = {
  __typename: 'Query',
  additionalProducts: additionalProducts,
  berthApplication: berthApplication,
  boatTypes: boatTypes,
};
