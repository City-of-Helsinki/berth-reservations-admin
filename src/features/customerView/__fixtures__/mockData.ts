import { OrderOrderType, OrderStatus, PriceUnits, ProductServiceType } from '../../../@types/__generated__/globalTypes';
import { BerthInvoice, WinterStorageInvoice } from '../types';
import { INDIVIDUAL_CUSTOMER_profile as CUSTOMER_PROFILE } from '../__generated__/INDIVIDUAL_CUSTOMER';

export const emptyMockProfile: CUSTOMER_PROFILE = {
  __typename: 'ProfileNode',
  berthApplications: null,
  berthLeases: null,
  winterStorageLeases: null,
  boats: null,
  comment: null,
  firstName: '',
  id: '',
  invoicingType: null,
  language: null,
  lastName: '',
  orders: null,
  organization: null,
  primaryAddress: null,
  primaryEmail: null,
  primaryPhone: null,
  customerGroup: null,
};

export const mockInvoices: (BerthInvoice | WinterStorageInvoice)[] = [
  {
    orderNumber: 'MOCK-INVOICE',
    orderType: OrderOrderType.LEASE_ORDER,
    status: OrderStatus.WAITING,
    berthInformation: {
      number: '1',
      pierIdentifier: 'C',
      harborName: 'Aurinkolahden venesatama (Aurinkoranta)',
    },
    contractPeriod: {
      startDate: '2020-06-26',
      endDate: '2020-09-14',
    },
    dueDate: '2020-07-10',
    basePrice: 96.67,
    totalPrice: 191.7,
    orderLines: [
      {
        product: ProductServiceType.ELECTRICITY,
        price: 20.5,
        priceUnit: PriceUnits.PERCENTAGE,
        priceValue: 9,
      },
    ],
  },
];
