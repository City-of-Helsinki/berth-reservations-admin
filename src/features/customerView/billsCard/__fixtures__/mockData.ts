import { ProductServiceType } from '../../../../@types/__generated__/globalTypes';

export const mockBills = [
  {
    berthInformation: {
      number: 1,
      pierIdentifier: 'C',
      harborName: 'Aurinkolahden venesatama (Aurinkoranta)',
    },
    contractPeriod: {
      startDate: '2020-06-26',
      endDate: '2020-09-14',
    },
    dueDate: '2020-07-10',
    basePrice: 96.67,
    basePriceTaxPercentage: 24.0,
    totalPrice: 191.7,
    totalPriceTaxPercentage: 19.36,
    orderLines: [
      {
        product: ProductServiceType.ELECTRICITY,
        price: 20.5,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.WATER,
        price: 19.33,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.GATE,
        price: 2.42,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.MOORING,
        price: 8.33,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.WASTE_COLLECTION,
        price: 4.17,
        taxPercentage: 24.0,
      },
      {
        product: ProductServiceType.LIGHTING,
        price: 40.28,
        taxPercentage: 24.0,
      },
    ],
  },
];
