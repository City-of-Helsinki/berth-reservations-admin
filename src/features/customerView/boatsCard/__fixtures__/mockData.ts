import { Boat, BoatCertificate, LargeBoat } from '../../types';
import { BoatCertificateType } from '../../../../@types/__generated__/globalTypes';

export const insuraceCert: BoatCertificate = {
  file: 'https://venepaikka-api.test.hel.ninja/media/boats/2c736de2-5659-46ea-ad3f-9c1364307d7d/dummy_pdf2.pdf',
  certificateType: BoatCertificateType.INSURANCE,
  validUntil: null,
  checkedAt: '2020-04-28',
  checkedBy: null,
};

export const inspectionCert: BoatCertificate = {
  file: null,
  certificateType: BoatCertificateType.INSPECTION,
  validUntil: '2099-09-04',
  checkedAt: '2020-04-28',
  checkedBy: 'Jukka Virtanen',
};

export const boat: Boat = {
  id: '1111',
  boatType: {
    id: '7',
    name: 'Troolari',
  },
  width: 2,
  length: 10,
  draught: 2,
  weight: 15000,
  name: 'Huh hah hei',
  model: 'Trawl 11',
  registrationNumber: 'TRLR-555',
};

export const largeBoat: LargeBoat = {
  id: '9999',
  boatType: {
    id: '8',
    name: 'Suuri alus (yli 20t)',
  },
  width: 4,
  length: 15,
  draught: 2,
  weight: 15000,
  name: 'DevilsDriver',
  model: 'Trawl 77',
  registrationNumber: 'TRLR-666',
  propulsion: 'Bensin',
  hullMaterial: 'Aluminium',
  intendedUse: 'For fun',
  certificates: [inspectionCert, insuraceCert],
};
