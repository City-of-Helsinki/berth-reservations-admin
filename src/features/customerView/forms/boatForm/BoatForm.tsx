import React, { useState } from 'react';
import { Formik } from 'formik';
import * as Yup from 'yup';
import { useTranslation } from 'react-i18next';
import { TFunction } from 'i18next';
import { ObjectSchema } from 'yup';
import { Button, TextInput } from 'hds-react';

import { BoatType, FormProps } from '../../../harborView/forms/types';
import { Boat } from '../../types';
import FormHeader from '../../../../common/formHeader/FormHeader';
import styles from './boatForm.module.scss';
import ConfirmationModal from '../../../../common/confirmationModal/ConfirmationModal';
import Select from '../../../../common/select/Select';

export interface BoatFormProps extends FormProps<Boat> {
  boatTypes: BoatType[];
}

const getBoatValidationSchema = (t: TFunction, boatTypes: BoatType[]): ObjectSchema => {
  return Yup.object().shape({
    boatType: Yup.object().shape({
      id: Yup.string()
        .oneOf(boatTypes.map((boatType) => boatType.id))
        .required(t('forms.common.errors.required')),
    }),
    registrationNumber: Yup.string().required(t('forms.common.errors.required')),
    width: Yup.number()
      .typeError(t('forms.common.errors.numberType'))
      .positive(t('forms.common.errors.positive'))
      .required(t('forms.common.errors.required')),
    length: Yup.number()
      .typeError(t('forms.common.errors.numberType'))
      .positive(t('forms.common.errors.positive'))
      .required(t('forms.common.errors.required')),
    draught: Yup.number().typeError(t('forms.common.errors.numberType')).positive(t('forms.common.errors.positive')),
    weight: Yup.number().typeError(t('forms.common.errors.numberType')).positive(t('forms.common.errors.positive')),
  });
};

const transformValues = (values: any): Boat => {
  const { width, length, draught, weight } = values;
  return {
    ...values,
    width: parseFloat(width),
    length: parseFloat(length),
    draught: parseFloat(draught),
    weight: parseFloat(weight),
  };
};

const BoatForm = ({ onCancel, onDelete, onSubmit, isSubmitting, initialValues, boatTypes }: BoatFormProps) => {
  const { t } = useTranslation();
  const validationSchema = getBoatValidationSchema(t, boatTypes);
  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);

  const initial = initialValues ?? ({} as Boat);
  return (
    <Formik
      initialValues={initial}
      onSubmit={(values) => onSubmit?.(transformValues(values))}
      validateOnBlur={false}
      validateOnChange={false}
      validationSchema={validationSchema}
    >
      {({ values, errors, handleChange, handleSubmit }) => (
        <form onSubmit={handleSubmit} className={styles.form}>
          <FormHeader
            title={t('forms.boat.title')}
            isSubmitting={isSubmitting}
            onDelete={onDelete ? () => setIsDeleteModalOpen(true) : undefined}
            onDeleteText={t('forms.boat.delete')}
          />

          <Select
            id="boatType.id"
            label={t('forms.boat.boatType')}
            value={values.boatType.id}
            options={boatTypes.map((boatType) => {
              return {
                label: boatType.name ?? '-',
                value: boatType.id,
              };
            })}
            onChange={handleChange}
            required
          />
          <TextInput
            id="name"
            onChange={handleChange}
            value={values.registrationNumber}
            label={t('forms.boat.registrationNumber')}
            invalid={!!errors.registrationNumber}
            helperText={errors.registrationNumber}
          />

          <hr />

          <div className={styles.horizontal}>
            <TextInput
              id="width"
              value={values.width ? String(values.width) : ''}
              onChange={handleChange}
              label={t('forms.boat.width')}
              invalid={!!errors.width}
              helperText={errors.width}
            />
            <TextInput
              id="length"
              value={values.length ? String(values.length) : ''}
              onChange={handleChange}
              label={t('forms.boat.length')}
              invalid={!!errors.length}
              helperText={errors.length}
            />
            <TextInput
              id="draught"
              onChange={handleChange}
              value={values.draught ? String(values.draught) : ''}
              label={t('forms.boat.draught')}
              invalid={!!errors.draught}
              helperText={errors.draught}
            />
            <TextInput
              id="weight"
              onChange={handleChange}
              value={values.weight ? String(values.weight) : ''}
              label={t('forms.boat.weight')}
              invalid={!!errors.weight}
              helperText={errors.weight}
            />
          </div>

          <hr />

          <TextInput
            id="name"
            onChange={handleChange}
            value={values.name}
            label={t('forms.boat.name')}
            invalid={!!errors.name}
            helperText={errors.name}
          />
          <TextInput
            id="model"
            onChange={handleChange}
            value={values.model}
            label={t('forms.boat.model')}
            invalid={!!errors.model}
            helperText={errors.model}
          />

          <div className={styles.formActionButtons}>
            <Button variant="secondary" theme="black" disabled={isSubmitting} color="supplementary" onClick={onCancel}>
              {t('forms.common.cancel')}
            </Button>
            <Button theme="coat" type="submit" disabled={isSubmitting}>
              {t('forms.common.update')}
            </Button>
          </div>

          <ConfirmationModal
            isOpen={isDeleteModalOpen}
            title={t('forms.boat.title')}
            infoText={t('forms.boat.deleteConfirmation.infoText')}
            onCancelText={t('forms.common.cancel')}
            onConfirmText={t('forms.boat.delete')}
            warningText={t('forms.boat.deleteConfirmation.warningText')}
            onCancel={() => setIsDeleteModalOpen(false)}
            onConfirm={() => onDelete?.(values)}
            className={styles.confirmationModal}
          />
        </form>
      )}
    </Formik>
  );
};

export default BoatForm;
