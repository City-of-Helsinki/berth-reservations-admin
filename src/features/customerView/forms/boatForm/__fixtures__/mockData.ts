import { BoatType } from '../../../../harborView/forms/types';
import { Boat } from '../../../types';

export const mockBoatTypes: BoatType[] = [{ id: '1', name: 'Purjevene' }];

export const mockValidValues: Boat = {
  boatType: mockBoatTypes[0],
  id: 'id',
  model: 'model',
  name: 'name',
  registrationNumber: '123456',
  weight: 1,
  width: 1,
  length: 1,
  draught: 1,
};

export const mockInvalidValues: Boat = {
  boatType: mockBoatTypes[0],
  id: 'id',
  model: 'model',
  name: 'name',
  registrationNumber: '',
  weight: -10,
  width: 1,
  length: 1,
  draught: 1,
};
