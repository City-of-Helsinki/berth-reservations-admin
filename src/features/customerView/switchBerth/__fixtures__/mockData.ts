import { HarborOption } from '../types';
import { SWITCH_BERTH_HARBORS } from '../__generated__/SWITCH_BERTH_HARBORS';

export const mockOption1: HarborOption = {
  label: 'Aataminsatama',
  value: '3',
};

export const mockOption2: HarborOption = {
  label: 'Albertinsatama',
  value: '2',
};

export const mockOption3: HarborOption = {
  label: 'Bertansatama',
  value: '4',
};

export const mockOption4: HarborOption = {
  label: 'Zorronsatama',
  value: '1',
};

export const mockSwitchBerthHarbors: SWITCH_BERTH_HARBORS = {
  harbors: {
    __typename: 'HarborNodeConnection',
    edges: [
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '1',
          properties: { __typename: 'HarborProperties', name: 'Zorronsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '2',
          properties: { __typename: 'HarborProperties', name: 'Albertinsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '3',
          properties: { __typename: 'HarborProperties', name: 'Aataminsatama' },
        },
      },
      {
        __typename: 'HarborNodeEdge',
        node: {
          __typename: 'HarborNode',
          id: '4',
          properties: { __typename: 'HarborProperties', name: 'Bertansatama' },
        },
      },
    ],
  },
};
