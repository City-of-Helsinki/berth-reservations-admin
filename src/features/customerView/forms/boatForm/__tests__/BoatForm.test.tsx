import React from 'react';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import BoatForm from '../BoatForm';
import { mockBoatTypes, mockInvalidValues, mockValidValues } from '../__fixtures__/mockData';

describe('BoatForm', () => {
  it('renders correctly', () => {
    const wrapper = mount(<BoatForm boatTypes={mockBoatTypes} initialValues={mockValidValues} />);
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('invalid values cannot be submitted', async () => {
    const onSubmitMock = jest.fn();
    const wrapper = mount(
      <BoatForm boatTypes={mockBoatTypes} initialValues={mockInvalidValues} onSubmit={onSubmitMock} />
    );
    await act(async () => {
      wrapper.find('form').simulate('submit');
    });
    expect(onSubmitMock.mock.calls.length).toEqual(0);
  });

  it('valid values can be submitted', async () => {
    const onSubmitMock = jest.fn();
    const wrapper = mount(
      <BoatForm boatTypes={mockBoatTypes} initialValues={mockValidValues} onSubmit={onSubmitMock} />
    );
    await act(async () => {
      wrapper.find('form').simulate('submit');
    });
    expect(onSubmitMock.mock.calls.length).toEqual(1);
  });
});
