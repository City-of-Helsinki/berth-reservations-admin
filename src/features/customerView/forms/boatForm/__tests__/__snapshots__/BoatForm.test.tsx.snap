// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoatForm renders correctly 1`] = `
<form
  class="form"
>
  <div
    class="heading"
  >
    <h4
      class="text brand h4 titleText"
    >
      Veneen tiedot
    </h4>
  </div>
  <div
    class="Dropdown-module_root__2LI4- select"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="boatType.id-toggle-button"
      id="boatType.id-label"
    >
      Veneen tyyppi
      <span
        class="RequiredIndicator-module_indicator__Pt7GX"
      >
        *
      </span>
    </label>
    <div
      class="Dropdown-module_wrapper__1oIw5"
    >
      <button
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="boatType.id-label boatType.id-toggle-button"
        class="Dropdown-module_buttonDropdown__IwbnQ Dropdown-module_dropdown__22ZdF"
        id="boatType.id-toggle-button"
        type="button"
      >
        <span
          class="Dropdown-module_buttonLabel__2jVRU"
        >
          Purjevene
        </span>
        <svg
          class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik Dropdown-module_angleIcon__1Y1mT"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
          >
            <path
              d="M0 0h24v24H0z"
            />
            <path
              d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
              fill="currentColor"
            />
          </g>
        </svg>
      </button>
    </div>
    <ul
      aria-labelledby="boatType.id-label"
      class="Dropdown-module_menu__3rei6"
      id="boatType.id-menu"
      role="listbox"
      style="max-height: calc(var(--dropdown-height) * 5);"
      tabindex="-1"
    />
  </div>
  <div
    class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="name"
    >
      Rekisterinumero
    </label>
    <div
      class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
    >
      <input
        class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
        id="name"
        type="text"
        value="123456"
      />
    </div>
  </div>
  <hr />
  <div
    class="horizontal"
  >
    <div
      class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="width"
      >
        Leveys (m)
      </label>
      <div
        class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
      >
        <input
          class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
          id="width"
          type="text"
          value="1"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="length"
      >
        Pituus (m)
      </label>
      <div
        class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
      >
        <input
          class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
          id="length"
          type="text"
          value="1"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="draught"
      >
        Syv√§ys (m)
      </label>
      <div
        class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
      >
        <input
          class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
          id="draught"
          type="text"
          value="1"
        />
      </div>
    </div>
    <div
      class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
    >
      <label
        class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
        for="weight"
      >
        Paino (kg)
      </label>
      <div
        class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
      >
        <input
          class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
          id="weight"
          type="text"
          value="1"
        />
      </div>
    </div>
  </div>
  <hr />
  <div
    class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="name"
    >
      Nimi
    </label>
    <div
      class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
    >
      <input
        class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
        id="name"
        type="text"
        value="name"
      />
    </div>
  </div>
  <div
    class="TextInput-module_root__1ceal text-input_hds-text-input__24bl6"
  >
    <label
      class="FieldLabel-module_label__2tl-4 text-input_hds-text-input__label__pDw_W "
      for="model"
    >
      Merkki
    </label>
    <div
      class="TextInput-module_inputWrapper__1ru3h text-input_hds-text-input__input-wrapper__2p7Uy"
    >
      <input
        class="TextInput-module_input__35GDy text-input_hds-text-input__input__2XAyI"
        id="model"
        type="text"
        value="model"
      />
    </div>
  </div>
  <div
    class="formActionButtons"
  >
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_secondary__399Vd button_hds-button--secondary__1NOWS Button-module_theme-black__zO4WO button_hds-button--theme-black__3K_k4"
      color="supplementary"
      type="button"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Peruuta
      </span>
    </button>
    <button
      class="Button-module_button__3wgee button_hds-button__2A0je Button-module_primary__3bo_T button_hds-button--primary__2NVvO Button-module_theme-coat__2xA5D button_hds-button--theme-coat__12cO0"
      type="submit"
    >
      <span
        class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
      >
        Tallenna
      </span>
    </button>
  </div>
</form>
`;
